### PR TITLE
feat(helm): enable/disable prometheus scrapping configuration in annotation

### PR DESCRIPTION
**Pull request by mathieuherbert**
### Description

feature flag in the helm chart to enable/disable prometheus scrapping configuration
### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by mathieuherbert** [comment]
@johngmyers @kishorj is it possible to have a review please ?

File: helm/aws-load-balancer-controller/templates/deployment.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ spec:
         {{- toYaml .Values.podLabels | nindent 8 }}
         {{- end }}
       annotations:
-        {{- if not .Values.serviceMonitor.enabled }}
+        {{- if .Values.prometheusScrapping.enabled }}
```
Comment (kishorj) [inline helm/aws-load-balancer-controller/templates/deployment.yaml:29]: You are effectively replacing the serviceMonitor.enabled with the new configuration. Did you meant either/or for those instead?

Comment (mathieuherbert) [reply helm/aws-load-balancer-controller/templates/deployment.yaml]: Hi, 
I added `and` operator

**Comment by mathieuherbert** [comment]
@kishorj is it possible to have another review please

File: helm/aws-load-balancer-controller/templates/deployment.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ spec:
         {{- toYaml .Values.podLabels | nindent 8 }}
         {{- end }}
       annotations:
-        {{- if not .Values.serviceMonitor.enabled }}
+        {{- if and (.Values.prometheusScrapping.enabled) (not .Values.serviceMonitor.enabled) }}
```
Comment (johngmyers) [inline helm/aws-load-balancer-controller/templates/deployment.yaml:29]:
```suggestion
        {{- if and (.Values.prometheusScraping.enabled) (not .Values.serviceMonitor.enabled) }}
```

File: helm/aws-load-balancer-controller/values.yaml [hunk 3]
```diff
@@ -326,6 +326,10 @@ serviceMonitor:
   # Namespace to create the service monitor in
   namespace:
 
+#prometheusScrapping enable Promehteus Scrapping configuration in annotation section
+prometheusScrapping:
```
Comment (johngmyers) [inline helm/aws-load-balancer-controller/values.yaml:330]:
```suggestion
#prometheusScraping enables Prometheus scraping configuration in annotation section
prometheusScraping:
```

File: helm/aws-load-balancer-controller/values.yaml [hunk 4]
```diff
@@ -326,6 +326,10 @@ serviceMonitor:
   # Namespace to create the service monitor in
   namespace:
 
+#prometheusScrapping enable Promehteus Scrapping configuration in annotation section
```
Comment (johngmyers) [inline helm/aws-load-balancer-controller/values.yaml:329]: This does not adequately describe the end-effect of the setting. It should describe the behavior, not the mechanism.

**Review by johngmyers — CHANGES_REQUESTED** [review]
So people who have currently configured a `serviceMonitor` will no longer get metrics unless they now also specify an additional configuration option?

Why would someone want a ServiceMonitor without having Prometheus scrape it?

Since this has no effect unless there is a `ServiceMonitor`, why is this setting not under the `serviceMontior` block?

**Pull request closed without merging**